modules/misc/src/materials/ADDensity.C

```diff
@@ -0,0 +1,137 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+#include "ADDensity.h"
+
+registerADMooseObject("MiscApp", ADDensity);
+
+defineADValidParams(
+    ADDensity, ADMaterial, params.addClassDescription("Creates density AD material property");
+    params.addCoupledVar("disp_r", "The r displacement");
+    params.addCoupledVar("disp_x", "The x displacement");
+    params.addCoupledVar("disp_y", "The y displacement");
+    params.addCoupledVar("disp_z", "The z displacement");
+    params.addCoupledVar(
+        "displacements",
+        "The displacements appropriate for the simulation geometry and coordinate system");
+    params.addRequiredParam<Real>("density", "Initial density"););
+
+template <ComputeStage compute_stage>
+ADDensity<compute_stage>::ADDensity(const InputParameters & parameters)
+  : ADMaterial<compute_stage>(parameters),
+    _is_coupled(true),
+    _disp_r(isCoupled("displacements")
+                ? adCoupledValue("displacements", 0)
+                : (isCoupled("disp_r") ? adCoupledValue("disp_r") : adZero())),
+    _initial_density(adGetParam<Real>("density")),
+    _density(adDeclareADProperty<Real>("density"))
+{
+  const bool is_r_coupled = isCoupled("disp_r");
+  const bool is_x_coupled = isCoupled("disp_x");
+  const bool is_y_coupled = isCoupled("disp_y");
+  const bool is_z_coupled = isCoupled("disp_z");
+
+  // new parameter scheme
+  if (isCoupled("displacements"))
+  {
+    // get coordinate system
+    _coord_system = getBlockCoordSystem();
+
+    // get coupled gradients
+    const unsigned int ndisp = coupledComponents("displacements");
+    _grad_disp.resize(ndisp);
+    for (unsigned int i = 0; i < ndisp; ++i)
+      _grad_disp[i] = &adCoupledGradient("displacements", i);
+
+    // fill remaining components with zero
+    _grad_disp.resize(3, &adGradZero());
+  }
+
+  // old deprecated parameters
+  else if (is_x_coupled || is_r_coupled)
```

Comment:
Since we're creating a new class and we have a clean slate here, I think we should either eliminate code that is "deprecated" or not call it deprecated

Reply:
I wanted to remove, but I know it's been pushed in the past to maintain all previous features. I'll rip this out to just use displacements.